drivers/spi/spi_andes_atcspi200.c

```diff
@@ -0,0 +1,629 @@
+#define REG_CMD         0x24
+#define REG_ADDR        0x28
+#define REG_DATA        0x2c
+#define REG_CTRL        0x30
+#define REG_STAT        0x34
+#define REG_INTEN       0x38
+#define REG_INTST       0x3c
+#define REG_TIMIN       0x40
+#define REG_MCTRL       0x50
+#define REG_SLVST       0x60
+#define REG_SDCNT       0x64
+#define REG_CONFIG      0x7c
+
+#define DEV_CFG(dev) ((const struct spi_atcspi200_cfg * const) (dev)->config)
+#define DEV_DATA(dev) ((struct spi_atcspi200_data *) (dev)->data)
+
+#define SPI_IDR(dev) (DEV_CFG(dev)->base + REG_IDR)
+#define SPI_TFMAT(dev) (DEV_CFG(dev)->base + REG_TFMAT)
+#define SPI_DIRIO(dev) (DEV_CFG(dev)->base + REG_DIRIO)
+#define SPI_TCTRL(dev) (DEV_CFG(dev)->base + REG_TCTRL)
+#define SPI_CMD(dev) (DEV_CFG(dev)->base + REG_CMD)
+#define SPI_ADDR(dev) (DEV_CFG(dev)->base + REG_ADDR)
+#define SPI_DATA(dev) (DEV_CFG(dev)->base + REG_DATA)
+#define SPI_CTRL(dev) (DEV_CFG(dev)->base + REG_CTRL)
+#define SPI_STAT(dev) (DEV_CFG(dev)->base + REG_STAT)
+#define SPI_INTEN(dev) (DEV_CFG(dev)->base + REG_INTEN)
+#define SPI_INTST(dev) (DEV_CFG(dev)->base + REG_INTST)
+#define SPI_TIMIN(dev) (DEV_CFG(dev)->base + REG_TIMIN)
+#define SPI_MCTRL(dev) (DEV_CFG(dev)->base + REG_MCTRL)
+#define SPI_SLVST(dev) (DEV_CFG(dev)->base + REG_SLVST)
+#define SPI_SDCNT(dev) (DEV_CFG(dev)->base + REG_SDCNT)
+#define SPI_CONFIG(dev) (DEV_CFG(dev)->base + REG_CONFIG)
+
+/* Field mask of SPI transfer format register */
+#define REG_TFMAT_CPHA_OFFSET (0)
+#define REG_TFMAT_CPOL_OFFSET (1)
+#define REG_TFMAT_SLVMODE_OFFSET (2)
+#define REG_TFMAT_LSB_OFFSET (3)
+#define REG_TFMAT_MOSI_BIDIR_OFFSET (4)
+#define REG_TFMAT_DATA_MERGE_OFFSET (7)
+#define REG_TFMAT_DATA_LEN_OFFSET (8)
+#define REG_TFMAT_ADDR_LEN_OFFSET (16)
+
+#define REG_TFMAT_CPHA_MSK (0x1 << REG_TFMAT_CPHA_OFFSET)
+#define REG_TFMAT_CPOL_MSK (0x1 << REG_TFMAT_CPOL_OFFSET)
+#define REG_TFMAT_SLVMODE_MSK (0x1 << REG_TFMAT_SLVMODE_OFFSET)
+#define REG_TFMAT_LSB_MSK (0x1 << REG_TFMAT_LSB_OFFSET)
+#define REG_TFMAT_MOSI_BIDIR_MSK (0x1 << REG_TFMAT_MOSI_BIDIR_OFFSET)
+#define REG_TFMAT_DATA_MERGE_MSK (0x1 << REG_TFMAT_DATA_MERGE_OFFSET)
+#define REG_TFMAT_DATA_LEN_MSK (0x1f << REG_TFMAT_DATA_LEN_OFFSET)
+#define REG_TFMAT_ADDR_LEN_MSK (0x3 << REG_TFMAT_ADDR_LEN_OFFSET)
+
+/* Field mask of SPI direct IO control register */
+#define REG_DIRIO_CS_OUTPUT_OFFSET (8)
+#define REG_DIRIO_CS_OUTPUT_EN_OFFSET (16)
+
+#define REG_DIRIO_CS_OUTPUT_MSK (0x1 << REG_DIRIO_CS_OUTPUT_OFFSET)
+#define REG_DIRIO_CS_OUTPUT_EN_MSK (0x1 << REG_DIRIO_CS_OUTPUT_EN_OFFSET)
+
+/* Field mask of SPI transfer control register */
+#define REG_TCTRL_RD_TCNT_OFFSET (0)
+#define REG_TCTRL_DUMMY_CNT_OFFSET (9)
+#define REG_TCTRL_TOKEN_VAL_OFFSET (11)
+#define REG_TCTRL_WR_TCNT_OFFSET (12)
+#define REG_TCTRL_TOKEN_EN_OFFSET (21)
+#define REG_TCTRL_TRNS_MODE_OFFSET (24)
+#define REG_TCTRL_ADDR_FMT_OFFSET (28)
+#define REG_TCTRL_ADDR_EN_OFFSET (29)
+#define REG_TCTRL_CMD_EN_OFFSET (30)
```

Comment:
Remove unneeded defines and move everything to an header file.

Reply:
OK, I see!
Thanks for your remind.